client/src/ui/organisms/top-navigation-main/index.tsx

```diff
@@ -47,12 +48,12 @@ export const TopNavigationMain = ({ isOpenOnMobile }) => {
         <span className="visually-hidden">Show search</span>
       </Button>
 
-      {ENABLE_PLUS &&
-        ((isAuthenticated && (
-          <>
-            <UserMenu />
-          </>
-        )) || <AuthContainer />)}
+      {(ENABLE_PLUS && userData && userData.isAuthenticated && (
+        <>
+          <UserMenu />
+        </>
+      )) ||
+        (plusAvailable && <AuthContainer />) || <></>}
```

Comment:
Not going to split hairs now but, I believe you could return `null`

```suggestion
        (plusAvailable && <AuthContainer />) || null}
```